src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -253,17 +280,30 @@ internal static AccessibilityKind FromXElement(XElement accessibilityElement)
             {
                 return new AccessibilityKind((Accessibility)Enum.Parse(typeof(Accessibility), accessibilityElement.Value));
             }
+
+            public override bool Equals(object obj)
+            {
+                return Equals(obj as AccessibilityKind);
+            }
+
+            public bool Equals(AccessibilityKind other)
+            {
+                return this.Accessibility == other.Accessibility;
```

Comment:
You don't test for `other == null` but it could be from your `object.Equals` override.